src/ui/src/js/controllers/request_view.js

```diff
@@ -33,6 +33,33 @@ export default function requestViewController(
     request) {
   $scope.service = RequestService;
 
+ $scope.download = function() {
+      var filename = $scope.data.id;
+      var output;
+      var outputType = $scope.data.output_type;
+      if($scope.data.output === undefined || $scope.data.output == null){
+        window.alert("No output to download");
+        return;
```

Comment:
This is a good check! How about we also hide the button until we have something to download? That way we can prevent the users from making this error (which is nicer than alerting them (which is nicer than breaking :smile: )) 

See if you can figure out how to make this happen from the AngularJS docs, but take a look at the "Pour it again" button for something similar if you get stuck.

Reply:
Will do, I did not think of hiding button which is better than an alert.